HeatStroke/StateMachine/StateMachine.cpp

```diff
@@ -165,7 +165,8 @@ void HeatStroke::StateMachine::Update(const float p_fDelta, const bool m_bUpdate
 	if (m_bUpdateStack)
 	{
 		// Update entire stack from bottom to top
-		StateStack::iterator it = m_mStateStack.begin(), end = m_mStateStack.end();
+		StateStack mLocalCopy(m_mStateStack);
+		StateStack::iterator it = mLocalCopy.begin(), end = mLocalCopy.end();
```

Comment:
Ah, I see. I guarantee there's a better way to do this though. I will accept but can you write in the backlog that this would be nice to improve?